.evergreen/static-check.sh

```diff
@@ -11,4 +11,4 @@ source java-config.sh
 echo "mongo-hibernate: static checking ..."
 
 ./gradlew -version
-./gradlew -PxmlReports.enabled=true --info -x test clean check
+./gradlew -PxmlReports.enabled=true --info -x test clean check compileJava
```

Comment:
The above missing gradle task is needed to go about `ErrorProne` and `NullAway` code quality check, both of which happens during `compileJava` phase which has been excluded by `-x test`.